### ISSUE TITLE
Receipt's builtin resources counters are parsed incorrectly

### ISSUE THREAD

**Issue by Mirko-von-Leipzig** [eqlabs/pathfinder#1237]
Receipt builtin resource counters are not parsed correctly. 

The gateway sends these as optional values which are only set if they are non-zero. We parse them as either all being present or none being present which is incorrect. Unfortunately our empty enum is also missing `deny_unknown_fields` which means the gateway's partially filled counters would be incorrectly parsed as the empty set.

This implies all receipts are incorrect wrt to these values. i.e. we lost the data here.

On a positive note this data isn't exposed or used anywhere currently, i.e. its not part of the rpc api and therefore has little to no actual effect.

This will change once p2p lands as there they are used.
